gamemodes/base/entities/entities/lua_run.lua

```diff
@@ -6,21 +6,62 @@ ENT.DisableDuplicator = true
 
 AccessorFunc( ENT, "m_bDefaultCode", "DefaultCode" )
 
+local ALLOWED_LUA, ALLOWED_MAPS = include( "cfc_block_luarun/config.lua" )
+local CURRENT_MAP_ALLOWED = ALLOWED_MAPS[game.GetMap()]
+
 function ENT:Initialize()
+
+	-- If the entity has its first spawnflag set, run the code automatically
+	if self:HasSpawnFlags( SF_LUA_RUN_ON_SPAWN ) then
+		self:RunCode( self, self, self:GetDefaultCode() )
+	end
+
 end
 
 function ENT:KeyValue( key, value )
+
+	if key == "Code" then
+		self:SetDefaultCode( value )
+	end
+
 end
 
 function ENT:SetupGlobals( activator, caller )
+
+	ACTIVATOR = activator
+	CALLER = caller
+
+	if IsValid( activator ) and activator:IsPlayer() then
+		TRIGGER_PLAYER = activator
+	end
+
 end
 
 function ENT:KillGlobals()
+
+	ACTIVATOR = nil
+	CALLER = nil
+	TRIGGER_PLAYER = nil
+
 end
 
 function ENT:RunCode( activator, caller, code )
+
+	if not CURRENT_MAP_ALLOWED or not ALLOWED_LUA[util.MD5( code )] then
```

Comment:
Should localize util.MD5 at the top at the file, small peformance boost and prevents the function from being changed and breaking this addon.
```lua
local util_MD5 = util.MD5

-- later in the code you use it as
util_MD5( "example" )
```

Reply:
👍
Totally makes sense except for the breaking part, but extra performance is always a plus.